go/extractor/cli/go-autobuilder/go-autobuilder.go

```diff
@@ -803,18 +871,19 @@ func compareVersions(v versionInfo) (msg, version string) {
 // Check the versions of Go found in the environment and in the `go.mod` file, and return a
 // version to install. If the version is the empty string then no installation is required.
```

Comment:
So at the moment this is...

| Requested > Installed \\/ | None | Unsupported old | Unsupported new | Supported |
| --- | --- | --- | --- | --- |
| **None** | Install max supported | Install min supported | No action **(!)** | Install requested |
| **Unsupported old** | Install max supported | Install min supported | No action **(!)** | Install requested |
| **Unsupported new** | Install max supported | Install min supported | No action | Install requested |
| **Supported** | No action | No action | No action **(!!)** | Install requested if newer than installed |

Suggest the cells highlighted with **(!)** should be install max supported, in keeping with our practice elsewhere of not intentionally upgrading to a too-new Go, but responding constructively to a too-old or missing environmental Go.

Suggest the cell highlighted with **(!!)** should be install max supported if newer than installed, since it would be odd to rule that have go 1.17, request go 1.20 ==> upgrade to 1.20; have go 1.17, request go 1.21 ==> try with go 1.17.

Reply:
You and @mbg seem to agree on this, so I'll make the change